Run a crawler from a yaml file

sunu opened the pull request:
Run a crawler from a yaml file and optionally a source file directory. CLI syntax looks like `memorious run-file test.yaml --src foo/test/`. `run-file` also takes optional arguments used by `run` like `--threads`, `--flush` etc.

sunu committed:
Run a crawler from a yaml file

sunu committed:
Merge branch 'master' into sunu/run-yaml-file

sunu committed:
Bit of refactoring

sunu committed:
Add some doc strings

pudo commented on memorious/logic/manager.py:
```diff
@@ -34,19 +34,23 @@ def load_path(self, path):
+                    self.crawlers[crawler.name] = crawler
+                    return crawler
+                except ValueError as ex:
+                    log.warn(
```
Why not log.exception? It prints both the desired message and a full stack trace

pudo commented on memorious/logic/manager.py:
```diff
@@ -34,19 +34,23 @@ def load_path(self, path):
-            for stage in crawler:
-                yield crawler, stage
+    def load_crawler(self, path):
+        if os.path.isfile(path):
```
Is path a `pathlib.Path`? Then we could get this a bit simpler.

sunu committed:
`--src` option loads Python source files from `src` directory only

Instead of letting the use specify a directory of python files, we only
load `src` directory relative to the crawler config yaml file when
running a crawler through the `run-file` command.
